lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -913,6 +938,7 @@ def _stop_opensearch(self) -> None:
 
         if self.opensearch.is_node_up():
             # 1. Add current node to the voting + alloc exclusions
+            # TODO upgrade: disable on upgrade?
             self.opensearch_exclusions.add_current()
```

Comment:
> if stopping for upgrade, should we not add unit to exclusions?
We should

> also should we be doing this on normal restarts? e.g. restarting for tls or ip change
We also should because we're essentially avoiding opensearch to consider this node for new shards allocation

Reply:
To summarize, upgrades go through the `stop` process which incorporates the allocation exclusion of the node.
The same stop process used in any operation requiring a restart, and which may not guarantee the same node comes back online due to an internal failure and whatnot. (TLS certificates rotation, IP changes etc...)
 
We do this for 2 reasons:
1. Prevent new shards fropm being allocated to the node
2. Trigger a shards relocation to other nodes (missing from our implementation the synchrnous wait for this to finish)

In short, we should exclude the currently upgrading node from the allocation. And since we're making use of `stop_opensearch` method, we're already doing it.

Reply:
Addressed on MM